keps/0013-external-specs.md

```diff
@@ -0,0 +1,162 @@
+  FrameworkVersion FrameworkVersion
+  // The Plan being executed
+  Plan Plan
+  // The Phase being executed
+  Phase Phase
+  // The step that is being currently run
+  Step Step
+  // The parameter values from the Instance Spec
+  Arguments map[string]string
+}
+```
+
+into a list of Kubernetes objects. Therefore, defining an `Engine` interface for a new application spec as follows would allow for using that spec in the execution of a plan.
+
+```go
+type Engine interface{
+  // Turns the input variables into a list of Kubernetes objects to apply
+  // for this step
+  Render(args StepArgument) ([]runtime.Objects, error)
+}
+```
+
+The `PlanExecution` controller would select the proper engine based on the `engine` keyword added to the `FrameworkVersion` spec.
+
+### Engine Spec in FrameworkVersion
+
+The Framework Version Spec would have a new field `Engine` that would specify the engine that should be used to Render steps into Kubernetes objects. Initial implementation will focus on adding one additional engine to support the execution of `Helm` charts.
+
+```go
+// FrameworkVersionSpec defines the desired state of FrameworkVersion.
+type FrameworkVersionSpec struct {
+	// +optional
+	Framework corev1.ObjectReference `json:"framework,omitempty"`
+
+  ...
+
+  // Define the engine for rendering.  Defaults to `kudo`
+  Engine EngineType `json:"engine,omitempty"`
+}
+
+type EngineType string
+
+const (
+  // Use default kudo engine for rendering
+  EngineTypeKUDO EngineType = "kudo"
+  // Use helm engine for rendering framework
+  EngineTypeHelm EngineType = "helm"
+)
+
+```
+
+#### KUDO Engine
+
+The functionality and rendering of KUDO frameworks will not change as a result of this KEP.
+
+#### Helm Engine
+
+When a FrameworkVersion is created from a Helm Chart, we will refer to this FrameworkVersion as a **Helm FrameworkVersion** to differentiate it from the underlying **Helm Chart** and a FrameworkVersion that uses the KUDO Engine.
+
+##### Installation of Helm FrameworkVersions
+
+Helm charts have a similar structure to KUDO Frameworks and the converstion, at installation time, of a Helm Chart into a Helm FrameworkVersion should be straightfoward.
+
+- The `Chart.yaml` will be converted:
+  - Framework.name = name
+  - FrameworkVersion.name = name-version
+  - Framework.spec.maintainers = maintainers
+  - Framework.spec.url = url
+  - FrameworkVersion.kubernetesVersion = kubeVersion
+  - FrameworkVersion.kudoVersion = 0.3.0
+- Files in the `templates` folder will be inserted into the `templates` field in the FrameworkVersion, exactly how is done in KUDO FrameworkVersions.
+- There will be no parameters defined explicitly in the FrameworkVersion.
+
+By default there will be a `deploy` plan with one `deploy` phase and one `deploy` step that will render all objects in the chart.
+
+### User Stories
+
+- Allow running of a Helm Chart as a KUDO Framework
+- Allow running of a CNAB bundle as a KUDO Framework
```

Comment:
could we maybe focus on helm only for now? maybe put the rest to future work?

Reply:
Sounds like a plan!